src/lib/Cropper.svelte

```diff
@@ -229,34 +233,57 @@
       restrictPosition
     )
 
-    dispatch('cropcomplete', {
+    oncropcomplete?.({
       percent: croppedAreaPercentages,
       pixels: croppedAreaPixels,
     })
   }
 
   // ------ Reactive statement ------
   //when aspect changes, we reset the cropperSize
-  $: if (imgEl) {
-    cropperSize = cropSize ? cropSize : helpers.getCropSize(imgEl.width, imgEl.height, aspect)
-  }
+  $effect(() => {
+    if (imgEl) {
+      cropperSize = cropSize ? cropSize : helpers.getCropSize(imgEl.width, imgEl.height, aspect)
+    }
+  })
 
   // when zoom changes, we recompute the cropped area
-  $: zoom && emitCropData()
+  $effect(() => {
+    if (zoom) {
+      emitCropData()
+    }
+  })
 
-  $: if(aspect) {
-    computeSizes()
-    emitCropData()
-  } 
+  // this variable is required to prevent the effect into entering an infinite loop (https://github.com/ValentinH/svelte-easy-crop/pull/59#discussion_r1896254867)
+  let lastAspect = $state(aspect)
+  $effect(() => {
+    if (aspect !== lastAspect) {
+      lastAspect = aspect
+      computeSizes()
+      emitCropData()
+    }
+  })
+
+  const containerAction: Action<HTMLDivElement> = node => {
+    $effect(() => {
+      node.addEventListener('touchstart', onTouchStart, { passive: false })
```

Comment:
While comparing with the React version of this lib, I noticed that we don't set passive to false for this touch start event so we can probably safely remove it and check if there is no warning in the console.